xnat_tools/dicom_export.py

```diff
@@ -34,150 +32,83 @@
     get_scan_ids,
 )
 
-_logger = logging.getLogger(__name__)
-
+from xnat_tools.logging import setup_logging
 
-def parse_args(args):
-    """Parse command line parameters
 
-    Args:
-      args ([str]): command line parameters as list of strings
-
-    Returns:
-      :obj:`argparse.Namespace`: command line parameters namespace
-    """
-    parser = argparse.ArgumentParser(
-        description="Dump XNAT Session into a BIDS friendly directory"
-    )
-    parser.add_argument("--host", default="https://bnc.brown.edu/xnat", help="Host")
-    parser.add_argument("-u", "--user", help="XNAT username", required=True)
-    parser.add_argument(
-        "-p",
-        "--password",
-        type=XNATPass,
-        help="XNAT password",
-        default=XNATPass.DEFAULT,
-    )
-    parser.add_argument("--session", help="Session ID", required=True)
-    parser.add_argument(
-        "--session_suffix",
-        help="Suffix of the session for BIDS e.g, 01. This will produce a sesstion label of sess-01",
-        required=True,
-        type=str,
-    )
-    parser.add_argument(
-        "--bids_root_dir", help="Root output directory for BIDS files", required=True
-    )
-    parser.add_argument(
-        "--bidsmap_file",
-        help="Bidsmap JSON file to correct sequence names",
-        required=False,
-        default="",
-    )
-    parser.add_argument(
-        "--seqlist",
-        help="List of sequences from XNAT to run if don't want to process all seuqences",
-        required=False,
-        default=[],
-        nargs="*",  # 0 or more values expected => creates a list
-        type=int,
-    )
-    parser.add_argument(
-        "--skiplist",
-        help="List of sequences from XNAT to SKIP. Accepts a list --skiplist 1 2 3",
-        required=False,
-        default=[],
-        nargs="*",  # 0 or more values expected => creates a list
-        type=int,
-    )
-    parser.add_argument(
-        "--log_id",
+_logger = logging.getLogger(__name__)
+app = typer.Typer()
+
+
+@app.command()
+def dicom_export(
+    session: str = typer.Argument(
+        ..., help="XNAT Session ID, that is the Accession # for an experiment."
+    ),
+    bids_root_dir: str = typer.Argument(
+        ..., help="Root output directory for exporting the files"
+    ),
```

Comment:
Is there no way to reuse these arguments via `typer`? Seems kinda wasteful to have them repeated 3 times throughout `xnat-tools`

Reply:
I'll take a look